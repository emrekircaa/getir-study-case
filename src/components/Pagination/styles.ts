import styled from "styled-components";
import { colors } from "../../theme";

export const PaginationContainer = styled.div`
  margin-top: 32px;
  ul {
    display: flex !important;
    justify-content: space-around;
    align-items: center;
    list-style: none;
    .previous,
    .next {
      width: auto !important;
      &.disabled {
        div,
        a {
          color: ${colors.gray500} !important;
        }
        path {
          fill: ${colors.gray500} !important;
        }
      }
      div,
      a {
        color: ${colors.primary} !important;
      }
      path {
        fill: ${colors.primary} !important;
      }
    }
    .rp-page {
      display: flex;
      justify-content: center;
      align-items: center;
      width: 32px;
      height: 40px;
      color: ${colors.gray500};
      font-size: 14px;
      font-style: normal;
      font-weight: 600;
      line-height: 16px;
      cursor: pointer;
      &.selected {
        background-color: ${colors.primary};
        border-radius: 2px;
        color: #fff;
      }
    }
  }
`;

export const StyledLabelText = styled.div`
  font-weight: 600;
  font-size: 14px;
  line-height: 16px;
  margin: 0 7px;
`;
export const StyledLabel = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  color: black;
  svg {
    fill: black;
  }

  &:hover {
    cursor: pointer;
    color: black;

    svg {
      fill: black;
    }
  }
`;
