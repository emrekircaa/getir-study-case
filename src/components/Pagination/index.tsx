/* eslint-disable @typescript-eslint/no-explicit-any */
import React from "react";
import * as styles from "./styles";
import RightArrow from "../../assets/icons/RightArrow";
import LeftArrow from "../../assets/icons/LeftArrow";
import ReactPaginate from "react-paginate";

interface PaginationProps {
  pageCount: number; // total page count
  selectedPageIndex?: number; // currently selected page index
  onSelectedPageIndexChange: () => void; // function to invoke on selecting new pagination page
}

export const Pagination: React.FC<PaginationProps> = ({
  pageCount,
  selectedPageIndex = 0,
  onSelectedPageIndexChange,
  ...rest
}) => {
  const handlePageClick = () => onSelectedPageIndexChange();

  const getLabel = React.useCallback((isNextPage: boolean) => {
    return (
      <styles.StyledLabel>
        {isNextPage ? "" : <RightArrow />}
        <styles.StyledLabelText>
          {isNextPage ? "Next" : "Prev"}
        </styles.StyledLabelText>
        {isNextPage ? <LeftArrow /> : ""}
      </styles.StyledLabel>
    );
  }, []);

  return (
    <styles.PaginationContainer data-testid="pagination">
      <ReactPaginate
        nextLabel={getLabel(true)}
        breakLinkClassName="rp-break-link"
        pageClassName="rp-page"
        breakClassName="rp-break"
        onPageChange={handlePageClick}
        selectedPageRel="active"
        pageRangeDisplayed={3}
        pageLinkClassName="rp-link"
        pageCount={pageCount}
        previousLabel={getLabel(false)}
        forcePage={selectedPageIndex}
        {...rest}
      />
    </styles.PaginationContainer>
  );
};
