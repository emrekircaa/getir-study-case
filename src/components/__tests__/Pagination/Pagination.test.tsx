import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Pagination } from "../../Pagination/index"; // Update the import path based on your actual structure

describe("Pagination component", () => {
  it("should match snapshot", () => {
    const { getByTestId } = render(
      <Pagination
        pageCount={10}
        selectedPageIndex={0}
        onSelectedPageIndexChange={jest.fn()}
      />
    );
    const comp = getByTestId("pagination");
    expect(comp).toMatchSnapshot();
  });
});
