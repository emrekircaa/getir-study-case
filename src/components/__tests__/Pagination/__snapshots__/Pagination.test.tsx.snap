// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component should match snapshot 1`] = `
<div
  class="sc-aXZVg kLGOtx"
  data-testid="pagination"
>
  <ul
    aria-label="Pagination"
    role="navigation"
  >
    <li
      class="previous disabled"
    >
      <a
        aria-disabled="true"
        aria-label="Previous page"
        class=" "
        rel="prev"
        role="button"
        tabindex="-1"
      >
        <div
          class="sc-eqUAAy bTmlgM"
        >
          <svg
            fill="none"
            height="24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5.067 12.333a.766.766 0 0 0 .192.28l6.127 6.13a.873.873 0 1 0 1.234-1.235l-4.63-4.632h10.135A.878.878 0 0 0 19 12a.878.878 0 0 0-.875-.876H7.98l4.63-4.633a.873.873 0 1 0-1.234-1.235l-6.126 6.13c-.08.08-.14.176-.193.281a.92.92 0 0 0 .009.666Z"
              fill="#1EA4CE"
              fill-rule="evenodd"
            />
          </svg>
          <div
            class="sc-gEvEer gkWgdc"
          >
            Prev
          </div>
        </div>
      </a>
    </li>
    <li
      class="rp-page selected"
    >
      <a
        aria-current="page"
        aria-label="Page 1 is your current page"
        class="rp-link"
        rel="active"
        role="button"
        tabindex="-1"
      >
        1
      </a>
    </li>
    <li
      class="rp-page"
    >
      <a
        aria-label="Page 2"
        class="rp-link"
        rel="next"
        role="button"
        tabindex="0"
      >
        2
      </a>
    </li>
    <li
      class="rp-page"
    >
      <a
        aria-label="Page 3"
        class="rp-link"
        role="button"
        tabindex="0"
      >
        3
      </a>
    </li>
    <li
      class="rp-break"
    >
      <a
        aria-label="Jump forward"
        class="rp-break-link"
        role="button"
        tabindex="0"
      >
        ...
      </a>
    </li>
    <li
      class="rp-page"
    >
      <a
        aria-label="Page 8"
        class="rp-link"
        role="button"
        tabindex="0"
      >
        8
      </a>
    </li>
    <li
      class="rp-page"
    >
      <a
        aria-label="Page 9"
        class="rp-link"
        role="button"
        tabindex="0"
      >
        9
      </a>
    </li>
    <li
      class="rp-page"
    >
      <a
        aria-label="Page 10"
        class="rp-link"
        role="button"
        tabindex="0"
      >
        10
      </a>
    </li>
    <li
      class="next"
    >
      <a
        aria-disabled="false"
        aria-label="Next page"
        class=""
        rel="next"
        role="button"
        tabindex="0"
      >
        <div
          class="sc-eqUAAy bTmlgM"
        >
          <div
            class="sc-gEvEer gkWgdc"
          >
            Next
          </div>
          <svg
            fill="none"
            height="14"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M13.933 6.667a.765.765 0 0 0-.192-.28L7.614.257A.873.873 0 1 0 6.38 1.491l4.63 4.632H.875A.878.878 0 0 0 0 7c0 .482.394.876.875.876H11.02l-4.63 4.633a.873.873 0 1 0 1.234 1.235l6.126-6.13c.08-.08.14-.176.193-.281a.92.92 0 0 0-.009-.666Z"
              fill="#697488"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </a>
    </li>
  </ul>
</div>
`;
